diff --git a/pdal/drivers.py b/pdal/drivers.py
index 54ea882..bf604ea 100644
--- a/pdal/drivers.py
+++ b/pdal/drivers.py
@@ -5,6 +5,8 @@ from typing import Callable, ClassVar, FrozenSet, Mapping, Optional, Sequence, T
 
 from .pipeline import Filter, Reader, Stage, Writer
 
+import shlex
+
 StreamableTypes: FrozenSet
 
 
@@ -59,16 +61,33 @@ class Driver:
 
 
 def inject_pdal_drivers() -> None:
-    drivers = json.loads(
-        subprocess.run(["pdal", "--drivers", "--showjson"], capture_output=True).stdout
-    )
-    options = dict(
-        json.loads(
-            subprocess.run(
-                ["pdal", "--options", "all", "--showjson"], capture_output=True
-            ).stdout
-        )
-    )
+
+    command = 'pdal --drivers --showjson'
+
+    p = subprocess.run(shlex.split(command),
+                       stdout = subprocess.PIPE,
+                       stderr=subprocess.PIPE)
+    output, error = p.communicate()
+    if error:
+        raise RuntimeError(f"Unable to run pdal --drivers with error '{error}'")
+
+    try:
+        drivers = json.loads(output.decode('utf-8'))
+    except json.decoder.JSONDecodeError:
+        print (type(output))
+        print (output.decode('utf-8'))
+        raise
+
+    command = 'pdal --options all --showjson'
+
+    p = subprocess.run(shlex.split(command),
+                       stdout = subprocess.PIPE,
+                       stderr=subprocess.PIPE)
+    output, error = p.communicate()
+    if error:
+        raise RuntimeError(f"Unable to run pdal --options with error '{error}'")
+
+    options = dict( json.loads( output.decode('utf-8') ))
     streamable = []
     for d in drivers:
         name = d["name"]
