diff --git a/pdal/drivers.py b/pdal/drivers.py
index 54ea882..797ba26 100644
--- a/pdal/drivers.py
+++ b/pdal/drivers.py
@@ -59,16 +59,18 @@ class Driver:
 
 
 def inject_pdal_drivers() -> None:
-    drivers = json.loads(
-        subprocess.run(["pdal", "--drivers", "--showjson"], capture_output=True).stdout
-    )
-    options = dict(
-        json.loads(
-            subprocess.run(
+    p = subprocess.run(["pdal", "--drivers", "--showjson"], capture_output=True).stdout
+    if p.stderr:
+        raise RuntimeError(f"Unable to run pdal --drivers with error '{p.stderr}'")
+    drivers = json.loads(p.stdout)
+
+    p = subprocess.run(
                 ["pdal", "--options", "all", "--showjson"], capture_output=True
-            ).stdout
-        )
-    )
+            )
+    if p.stderr:
+        raise RuntimeError(f"Unable to run pdal --options with error '{p.stderr}'")
+
+    options = dict( json.loads( p.stdout ))
     streamable = []
     for d in drivers:
         name = d["name"]
