diff --git a/pdal/drivers.py b/pdal/drivers.py
index 54ea882..806c226 100644
--- a/pdal/drivers.py
+++ b/pdal/drivers.py
@@ -5,6 +5,8 @@ from typing import Callable, ClassVar, FrozenSet, Mapping, Optional, Sequence, T
 
 from .pipeline import Filter, Reader, Stage, Writer
 
+import shlex
+
 StreamableTypes: FrozenSet
 
 
@@ -59,16 +61,39 @@ class Driver:
 
 
 def inject_pdal_drivers() -> None:
-    drivers = json.loads(
-        subprocess.run(["pdal", "--drivers", "--showjson"], capture_output=True).stdout
-    )
-    options = dict(
-        json.loads(
-            subprocess.run(
-                ["pdal", "--options", "all", "--showjson"], capture_output=True
-            ).stdout
-        )
-    )
+
+    command = 'pdal --drivers --showjson'
+
+    p = subprocess.run(shlex.split(command),
+                    encoding='utf-8',
+                    capture_output=True)
+    output, error = p.stdout, p.stderr
+    if p.returncode:
+        print (f'return code {p.returncode} error: "{error}"')
+        if error:
+            raise RuntimeError(f"Unable to run pdal --drivers with error '{error}'")
+
+    try:
+        drivers = json.loads(output)
+    except json.decoder.JSONDecodeError as e:
+        print (type(output))
+        print (f'{e}')
+
+        print (output.decode('utf-8'))
+        raise
+
+    command = 'pdal --options all --showjson'
+
+    p = subprocess.run(shlex.split(command),
+                    encoding='utf-8',
+                    capture_output=True)
+    output, error = p.stdout, p.stderr
+    if p.returncode:
+        print (f'return code {p.returncode} error: "{error}"')
+        if error:
+            raise RuntimeError(f"Unable to run pdal --options with error '{error}'")
+
+    options = dict( json.loads( output ))
     streamable = []
     for d in drivers:
         name = d["name"]
